src/test/java/uk/gov/companieshouse/itemhandler/kafka/OrdersKafkaConsumerIntegrationErrorModeTest.java

```diff
@@ -0,0 +1,162 @@
+    private static final String ORDER_RECEIVED_TOPIC_RETRY = "order-received-retry";
+    private static final String ORDER_RECEIVED_TOPIC_ERROR = "order-received-error";
+    private static final String CONSUMER_GROUP_MAIN_RETRY = "order-received-main-retry";
+    private static final String CONSUMER_GROUP_MAIN_ERROR = "order-received-error";
+    private static final String ORDER_RECEIVED_URI = "/order/ORDER-12345";
+    @Value("${kafka.broker.addresses}")
+    private String brokerAddresses;
+
+    private KafkaTemplate<String, String> template;
+
+    private KafkaMessageListenerContainer<String, String> container;
+
+    private BlockingQueue<ConsumerRecord<String, String>> records;
+
+    @Autowired
+    private OrdersKafkaConsumerWrapper consumerWrapper;
+
+    @Before
+    public void setUp() {
+        setUpTestKafkaOrdersProducer();
+        setUpTestKafkaOrdersConsumer();
+    }
+
+    @After
+    public void tearDown() {
+        consumerWrapper.reset();
+        container.stop();
+    }
+
+    private void setUpTestKafkaOrdersProducer() {
+        final Map<String, Object> senderProperties = KafkaTestUtils.senderProps(brokerAddresses);
+
+        senderProperties.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+
+        final ProducerFactory<String, String> producerFactory = new DefaultKafkaProducerFactory<>(senderProperties);
+
+        template = new KafkaTemplate<>(producerFactory);
+        template.setDefaultTopic(ORDER_RECEIVED_TOPIC);
+    }
+
+    private void setUpTestKafkaOrdersConsumer() {
+        final Map<String, Object> consumerProperties = new HashMap<>();
+        consumerProperties.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
+        consumerProperties.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
+        consumerProperties.put(ConsumerConfig.GROUP_ID_CONFIG, CONSUMER_GROUP_MAIN_RETRY);
+        consumerProperties.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, brokerAddresses);
+
+        final DefaultKafkaConsumerFactory<String, String> consumerFactory =
+                new DefaultKafkaConsumerFactory<>(consumerProperties);
+
+        final ContainerProperties containerProperties = new ContainerProperties(ORDER_RECEIVED_TOPIC);
+
+        container = new KafkaMessageListenerContainer<>(consumerFactory, containerProperties);
+
+        records = new LinkedBlockingQueue<>();
+
+        container.setupMessageListener((MessageListener<String, String>) record -> {
+                    records.add(record);
+                });
+
+        container.start();
+
+        ContainerTestUtils.waitForAssignment(container, 0);
+    }
+
+    @Test
+    public void testOrdersConsumerReceivesOrderReceived1Message() throws InterruptedException, ExecutionException {
+        // When
+        final ListenableFuture<SendResult<String, String>> future =
+                template.send(ORDER_RECEIVED_TOPIC, ORDER_RECEIVED_URI);
+
```

Comment:
`future` not used in this test (applies to other tests in this file too).

Reply:
Removed